### PR TITLE
feat: implement vote deeplinking

### PR DESCRIPTION
**Pull request by goga-m**
## Summary
Closes https://app.clickup.com/t/2p89rt6

Basic scenarios covered:

* Vote delegate by name (boldninja) 
   https://arkvault-git-feat-vote-deeplinking-functionality-ardent.vercel.app/#/?method=vote&coin=ARK&nethash=2a44f340d76ffc3df204c5f38cd355b7496c9065a1ade2ef92071436bd72e867&delegate=boldninja
* Vote delegate by public key (arktoshi)
   https://arkvault-git-feat-vote-deeplinking-functionality-ardent.vercel.app/#/?method=vote&coin=ARK&nethash=2a44f340d76ffc3df204c5f38cd355b7496c9065a1ade2ef92071436bd72e867&publicKey=029918d8fe6a78cc01bbab31f636494568dd954431f75f4ea6ff1da040b7063a70

* Missing delegate info error (error toast requiring name or public key)
  https://arkvault-git-feat-vote-deeplinking-functionality-ardent.vercel.app/#/?method=vote&coin=ARK&nethash=2a44f340d76ffc3df204c5f38cd355b7496c9065a1ade2ef92071436bd72e867

* Unable to find a delegate by name (error toast)
  https://arkvault-git-feat-vote-deeplinking-functionality-ardent.vercel.app/#/?method=vote&coin=ARK&nethash=2a44f340d76ffc3df204c5f38cd355b7496c9065a1ade2ef92071436bd72e867&delegate=uknowndelegate
  
* Unable to find a delegate by public key (error toast)
  https://arkvault-git-feat-vote-deeplinking-functionality-ardent.vercel.app/#/?method=vote&coin=ARK&nethash=2a44f340d76ffc3df204c5f38cd355b7496c9065a1ade2ef92071436bd72e867&publicKey=1
  
* Require either delegate or public key (error toast that prompts to enter either a name or public key but not both)
  https://arkvault-git-feat-vote-deeplinking-functionality-ardent.vercel.app/#/?method=vote&coin=ARK&nethash=2a44f340d76ffc3df204c5f38cd355b7496c9065a1ade2ef92071436bd72e867&delegate=boldninja&publicKey=029918d8fe6a78cc01bbab31f636494568dd954431f75f4ea6ff1da040b7063a70

*  Show error for resigned delegate
   https://arkvault-git-feat-vote-deeplinking-functionality-ardent.vercel.app/#/?method=vote&coin=ARK&nethash=2a44f340d76ffc3df204c5f38cd355b7496c9065a1ade2ef92071436bd72e867&delegate=testdelegate
  

## Checklist

-   [x] My changes look good in both light AND dark mode
-   [x] The change is not hardcoded to a single network, but has multi-asset in mind
-   [x] I checked my changes for obvious issues, debug statements and commented code
-   [ ] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

File: src/app/hooks/use-search-parameters-validation.ts [hunk 1]
```diff
@@ -86,8 +87,18 @@ export const useSearchParametersValidation = () => {
 
 		await env.delegates().sync(profile, network.coin(), network.id());
 
-		if (!delegateFromSearchParameters({ env, network, profile, searchParameters: parameters })) {
-			throw new Error(t("TRANSACTION.VALIDATION.DELEGATE_NOT_FOUND", { delegate: delegateName || publicKey }));
+		const delegate = delegateFromSearchParameters({ env, network, profile, searchParameters: parameters });
+
+		const truncatedName = truncate(delegateName || (publicKey as string), {
```
Comment (dated) [inline src/app/hooks/use-search-parameters-validation.ts:92]: should we truncate only if the value is longer than 20 chars, i.e. only when a public key is used?

Comment (goga-m) [reply src/app/hooks/use-search-parameters-validation.ts]: Fixed @dated, thanks :+1:

**Review by samharperpittam — APPROVED** [review]

File: src/domains/transaction/pages/SendVote/SendVote.tsx [hunk 2]
```diff
@@ -135,8 +184,9 @@ export const SendVote = () => {
 
 			appendParameters(parameters, "vote", voteDelegates);
 
+			//	TODO: redirect to general votes page if wallet is not defined.
```
Comment (ItsANameToo) [inline src/domains/transaction/pages/SendVote/SendVote.tsx:187]: 🤔

Comment (goga-m) [reply src/domains/transaction/pages/SendVote/SendVote.tsx]: @ItsANameToo fixed. Redirecting to dashboard if the user comes from a deeplink url. Thanks!

**Review by dated — APPROVED** [review]

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**